Core/DataAccess/EntityFramework/EfEntityRepositoryBase.cs

```diff
@@ -0,0 +1,59 @@
+﻿using Microsoft.EntityFrameworkCore;
+using System.Linq.Expressions;
+using WebNetSample.Core.Entities;
+
+namespace Core.DataAccess.EntityFramework
+{
+    public class EfEntityRepositoryBase<TEntity, TContext> : IEntityRepository<TEntity>
+         where TEntity : BaseEntity, new()
+         where TContext : DbContext, new()
+    {
+        public void Add(TEntity entity)
+        {
+            using (var context = new TContext())
```

Comment:
Isnt it better to close/destroy connection/instances after we completed the process?

Reply:
Well, it is not too good since in scope of one request you can easily make calls to DB several times, reopening new connections is not a cheap operation and could take some time, especially, if there are lot of such requests